### PR TITLE
Add testcase for assign role to LDAP group.

### PR DESCRIPTION
**Pull request by stonezdj**
Including
1. LDAP group configure settings
2. Assign role to LDAP user group.
3. Manage user group

### CODE REVIEW

File: tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    1. ldap_group_filter
+    1. ldap_gid 
+    1. ldap_group_scope   
+* A linux host with Docker CLI installed (Docker client).
+
+# Test Steps:
+
+**NOTE:**  
+* Create group  harbor_guest, harbor_dev, harbor_admin in LDAP.
+* Create LDAP user guest_user, dev_user, admin_user in harbor.
```
Comment (reasonerjt) [inline tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md]: is ldap or harbor?

File: tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* Assign add group following members
+    1. harbbor_guest --- guest_user, admin_user.
+    1. harbor_dev    --- dev_user.
+    1. harbor_admin  --- admin_user.
+
+* Login as admin user, create project private proj_group_test
```
Comment (reasonerjt) [inline tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md]: private project？

File: tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+* Add following group with the roles to proj_group_test
+    1. harbor_guest  --- guest, add this member with LDAP Group DN directly: cn=harbor_guest,ou=groups,dc=example,dc=com.
+    2. harbor_dev    --- developer, create user group with LDAP group DN directly: cn=harbor_developer,ou=groups,dc=example,dc=com
+    3. Add a user group: group DN: cn=harbor_admin,ou=groups,dc=example,dc=com, with name harbor_admin, 
+    4. Add project member, select existing user group harbor_admin, assign role administrator.
+* Login user guest_user, dev_user, admin_user in web console. all of them can see the proj_group_test.
+* Login user guest_user, dev_user, admin_user in command line. try to push pull images.
+* Remove admin_user from LDAP group harbor_admin, login again with admin_user. check the role in project proj_group_test
```
Comment (reasonerjt) [inline tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md]: This step and the following step are manual only right?  can we automate these steps?

File: tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+* Remove admin_user from LDAP group harbor_guest, login again with admin_user, check the role in project 
+proj_group_test
+
+
+# Expected Results:
+
```
Comment (reasonerjt) [inline tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md]: Would it be better if we put the expected result into test steps?  so it's easier for executing and automate ?

File: tests/testcases/Group3-RBAC/3-09-LDAP-usergroup-update.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+Test 3-9 LDAP usergroup update
```
Comment (reasonerjt) [inline tests/testcases/Group3-RBAC/3-09-LDAP-usergroup-update.md]: Do we need to support it on UI?

File: tests/testcases/Group3-RBAC/3-08-LDAP-usergroup-add.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+Test 3-8 LDAP Usergroup Add
```
Comment (reasonerjt) [inline tests/testcases/Group3-RBAC/3-08-LDAP-usergroup-add.md]: seems 3-8, 3-9, 3-10 should belong to another group, `group management`

File: tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md (hunk 3 above)
Comment (stonezdj) [reply tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md]: This testcase can be automated

File: tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md (hunk 1 above)
Comment (stonezdj) [reply tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md]: change to LDAP

File: tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md (hunk 4 above)
Comment (stonezdj) [reply tests/testcases/Group3-RBAC/3-07-LDAP-usergroup-manage-project-groupmembers.md]: updated

File: tests/testcases/Group12-group-management/12-01-LDAP-usergroup-add.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+# References:
+
+User guide
+
+# Environments:
+
+* This test requires that a Harbor instance is running and available.
+* An LDAP server is running and available, and enabled memberof overlay feature
+* LDAP group config parameter are configured.
+    1. ldap_group_basedn
```
Comment (reasonerjt) [inline tests/testcases/Group12-group-management/12-01-LDAP-usergroup-add.md]: Minor issue the numbers are all "1"...

**Comment by reasonerjt** [comment]
@stonezdj please fix the number issue and merge the PR.

File: tests/testcases/Group12-group-management/12-01-LDAP-usergroup-add.md (hunk 7 above)
Comment (reasonerjt) [reply tests/testcases/Group12-group-management/12-01-LDAP-usergroup-add.md]: Never mind, it's the mark down for `i` `ii`

**Pull request merged**